pkg/volume/projected/projected_test.go

```diff
@@ -258,8 +258,8 @@ func TestCollectDataWithSecret(t *testing.T) {
 		testPodUID := types.UID("test_pod_uid")
 		pod := &v1.Pod{ObjectMeta: metav1.ObjectMeta{Namespace: testNamespace, UID: testPodUID}}
 		client := fake.NewSimpleClientset(tc.secret)
-		_, host := newTestHost(t, client)
-
+		tempDir, host := newTestHost(t, client)
```

Comment:
Can we use `t.Run(tc.name, func(t *testing.T)` to iterate over the test cases?  That way all variables created are enclosed in a single run and get cleaned up after each run instead of only at the very end of all the test cases.

Reply:
done, thanks